cachi2/core/models/sbom.py

```diff
@@ -335,6 +425,217 @@ def __hash__(self) -> int:
         )
 
 
+def _unique(elements: Iterable) -> list:
+    out, seen = [], set()
+    for el in elements:
+        if el in seen:
+            continue
+        seen.add(el)
+        out.append(el)
+    return out
+
```

Comment:
Why do we need this function over plain `set` usage? `SPDXRelation` is hashable (but not immutable which we should fix?), so what am I missing, especially since right before the call there's the following commentary:

>  \# Relationships, on the other hand, are amended, so new
>  \# relationships will be constructed. Further, identical
>  \# relationships should be dropped. Deduplication based on building
>  \# a set is considered safe because all fields of all elements are
>  \# used to compute a hash.
even though it's related to the `self.retarget_and_prune_relationships` call that follows it.

Reply:
We don't need it. Removed.